runtime/style-loader.ts

```diff
@@ -0,0 +1,35 @@
+import * as FontFaceObserver from 'fontfaceobserver';
+
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class SkyAppStyleLoader {
+  public static readonly LOAD_TIMEOUT: number = 3000;
+  public isLoaded: boolean = false;
+
+  public loadStyles(): Promise<any> {
+    const fontAwesome = new FontFaceObserver('FontAwesome');
+    const openSans = new FontFaceObserver('Open Sans');
+    const oswald = new FontFaceObserver('Oswald');
+
+    return Promise
+      .all([
+        // Specify a character for FontAwesome since some browsers will fail to detect
+        // when the font is loaded unless a known character with a different width
+        // than the default is not specified.
+        fontAwesome.load('\uf0fc', SkyAppStyleLoader.LOAD_TIMEOUT),
+        openSans.load(undefined, SkyAppStyleLoader.LOAD_TIMEOUT),
+        oswald.load(undefined, SkyAppStyleLoader.LOAD_TIMEOUT)
+      ])
+      .then(() => {
+        this.isLoaded = true;
+      })
+      .catch((error) => {
+        // Errors loading the font should not stop the page from rendering.
+        // Passing the error along in case the client wants to do something with it.
+        return Promise.resolve({
+          error: error
```

Comment:
Well, the "error" will still fire if the font is taking too long to load (so, it might still exist, but is taking longer than 3 seconds to load). We can console log it if that helps, but thought it would be helpful to provide the error to the consuming service, if they wanted to check for it. Otherwise, there's no way for the consumer to "know" that an error occurred, because it will always be resolved:
https://github.com/blackbaud/skyux-builder/pull/166/files#diff-df8c52ab04ab7a3e6b86e304a41fd060R30